### ISSUE TITLE
🏆🥇 CompilerGym Code Optimization Competition finishes June 5! 🥇🏆

### ISSUE THREAD

**Issue by ChrisCummins** [facebookresearch/CompilerGym#681]
**tldr;** We are running the first code optimization competition at [MLArchSys'22](https://sites.google.com/view/mlarchsys/isca-2022/compiler-optimization-competition)! All are welcome to participate. Details [here](https://sites.google.com/view/mlarchsys/isca-2022/compiler-optimization-competition).

# Important Dates
- Competition Starts: ~~March 21, 2022~~
- Test Set Release: ~~May 17th, 2022~~
- Submission deadline: ~~May 31st~~ **June 5th, 2022 - Midnight Anywhere on Earth**
- Workshop: [June 19th, 2022](https://www.iscaconf.org/isca2022/program/workshops.php)

# Call for Participation

The optimization decisions that compilers make are critical for the performance and efficiency of compiled software. However, tuning and optimizing compilers is hard. There are many decisions to make, a near infinite number of possible programs, and most programs respond differently to optimizations. Performing this task by hand is time consuming, error prone, and yet often leads to suboptimal results. As a result, interest in applying AI techniques to this task is rapidly increasing.

The goal of this competition is to develop an agent for a classic problem in compilation, phase ordering to minimize code size. The phase ordering problem is to select, from a candidate set of optimization passes, which to run on an unseen input so as to produce the best output. Simple black box optimizations achieve strong results in minutes to hours of compute. The challenge is to achieve similar or smaller code size in less time. We provide training sets, starter code, and scripts to evaluate and validate results. The task is well-defined and suitable for a range of AI approaches. There is a broad scope of research opportunities, from gradient-free optimization to deep reinforcement learning.

# Submission Instructions

See the [competition site](https://sites.google.com/view/mlarchsys/isca-2022/compiler-optimization-competition) for submission instructions.

**Comment by ChrisCummins** [comment]
The evaluation script has been released and we are now open for your submissions! The deadline to submit is ~~May 31st~~ **June 5th, Midnight Anywhere on Earth**. Good luck!
